tests/testthat/test-bind.R

```diff
@@ -712,11 +712,63 @@ test_that("vec_rbind() fails with complex foreign S4 classes", {
 })
 
 test_that("vec_rbind() falls back to c() if S3 method is available", {
-  skip("TODO")
+  x <- foobar(1, foo = 1)
+  y <- foobar(2, bar = 2)
+
+  x_df <- data_frame(x = x)
+  y_df <- data_frame(x = y)
+
+  expect_error(vec_rbind(x_df, y_df), class = "vctrs_error_incompatible_type")
+
+  out <- with_methods(
+    c.vctrs_foobar = function(...) quux(NextMethod()),
+    vec_rbind(x_df, y_df)
+  )
+  expect_identical(out, data_frame(x = quux(c(1, 2))))
+
+  foo_df <- foobaz(x_df)
+  bar_df <- foobaz(y_df)
+
+  out <- with_methods(
+    c.vctrs_foobar = function(...) quux(NextMethod()),
+    vec_rbind(foo_df, bar_df)
+  )
+  expect_identical(out, foobaz(data_frame(x = quux(c(1, 2)))))
+
+  out <- with_methods(
```

Comment:
What's this testing? From the methods I expected it to be checking the `vec_ptype2` beats `c`, but methods are for different classes?

Reply:
The `foobaz()` is a `data.frame` subclass which contains a `foobar()` with a `c()` fallback. I've added a comment.